### PR TITLE
Fix Flow in GUI & Napari: key, channel, LMC in GUI; multi-channel UNet layers in Napari

### PR DESCRIPTION
**Pull request by qin-yu**
This PR attempts to fix:
- #173 

## Improvements on Napari Pipeline

For multi-channel UNets, return each channel of the prediction as a layer in Napari. I made minimal changes to detect and handle multi-channel output in `unet_predictions()`, `start_prediction_threading_process()` and `widget_unet_predictions()`.

image
Figure 1: an overlay of segmentation + boundary pred from multi-channel output + raw

## Improvements on Legacy GUI Pipeline

1. Add key and channel selection in all steps for GUI pipeline
2. Add LiftedMulticut (LMC) in GUI pipeline's segmentation frame, and add path/key/channel selection in it
3. Add key and channel default values in GUI config template

image
Figure 2: Key/channel selection for all steps and new LMC interface

### Design decisions

1. Tk GUI only allows fixed-typed entries. As a result, `None` key and `None` values has to be `"None"` string and `-1` integer respectively. This is changed back to `None` null value in GUI's main function and not handled by the `raw2seg` pipeline. The advantage of doing so is to have a centralised place for handling pipeline-specific constraints inside this pipeline's main. 
2. I usually stick to the original code format, but in widget file for GUI, GUI's Entry objects are formatted by Black formatter and I decided not to change back.

### Tests

I manually tested the following segmentations:

1. Ovule cell wall channel from multi-dataset HDF5 file as input, generic confocal 3D builtin as model, ovule nucleus label from multi-dataset as seeds for LMC
2. Ovule nucleus stain channel from multi-dataset HDF5 file as input, generic plant nuclei builtin as model (2-channel output), ovule nucleus label as seeds for LMC
3. Ovule nucleus stain channel from multi-dataset HDF5 file as input, generic plant nuclei builtin as model (2-channel output), ovule nucleus probability map (i.e. directly using the pre-saved channel 0) as non-seg seeds for LMC (**FAIL: but I don't think it's a problem of my changes**)
<details>

<summary>Error from vigra (if pmap is used for LMC)</summary>

```pytb
2024-01-31 10:45:55,293 [ThreadPoolExecutor-0_0] INFO PlantSeg - Computing segmentation with dtWS...
2024-01-31 10:55:53,307 [ThreadPoolExecutor-0_0] INFO PlantSeg - Clustering with LiftedMulticut...
2024-01-31 10:56:03,020 [ThreadPoolExecutor-0_0] ERROR PlantSeg - Python argument types in
    vigra.analysis.extractRegionFeatures(numpy.ndarray, numpy.ndarray)
did not match C++ signature:
    extractRegionFeatures(vigra::NumpyArray<3u, vigra::Singleband<float>, vigra::StridedArrayTag> volume, vigra::NumpyArray<3u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object histogramRange='globalminmax', int binCount=64, boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<2u, vigra::Singleband<float>, vigra::StridedArrayTag> image, vigra::NumpyArray<2u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object histogramRange='globalminmax', int binCount=64, boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<3u, vigra::TinyVector<float, 3>, vigra::StridedArrayTag> volume, vigra::NumpyArray<3u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<2u, vigra::TinyVector<float, 3>, vigra::StridedArrayTag> image, vigra::NumpyArray<2u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<4u, vigra::Multiband<float>, vigra::StridedArrayTag> volume, vigra::NumpyArray<3u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<3u, vigra::Multiband<float>, vigra::StridedArrayTag> image, vigra::NumpyArray<2u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
Traceback (most recent call last):
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/pipeline/executor.py", line 41, in _done_callback
    f.result()
  File "/g/kreshuk/yu/miniconda3/envs/plantseg-dev-gpu/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/miniconda3/envs/plantseg-dev-gpu/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/g/kreshuk/yu/miniconda3/envs/plantseg-dev-gpu/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/pipeline/raw2seg.py", line 118, in raw2seg
    output_paths = pipeline_step()
                   ^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/pipeline/steps.py", line 58, in __call__
    return [self.read_process_write(input_path) for input_path in self.input_paths]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/pipeline/steps.py", line 58, in <listcomp>
    return [self.read_process_write(input_path) for input_path in self.input_paths]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/segmentation/lmc.py", line 112, in read_process_write
    output_data = self.process(pmaps)
                  ^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/segmentation/lmc.py", line 83, in process
    segmentation = lifted_multicut_from_nuclei_pmaps(boundary_pmaps,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/repositories/plant-seg-doc/plantseg/segmentation/functional/segmentation.py", line 243, in lifted_multicut_from_nuclei_pmaps
    lifted_uvs, lifted_costs = lifted_problem_from_probabilities(rag, superpixels,
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/g/kreshuk/yu/miniconda3/envs/plantseg-dev-gpu/lib/python3.11/site-packages/elf/segmentation/features.py", line 501, in lifted_problem_from_probabilities
    mean_prob = vigra.analysis.extractRegionFeatures(inp, watershed, features=["mean"])["mean"]
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Boost.Python.ArgumentError: Python argument types in
    vigra.analysis.extractRegionFeatures(numpy.ndarray, numpy.ndarray)
did not match C++ signature:
    extractRegionFeatures(vigra::NumpyArray<3u, vigra::Singleband<float>, vigra::StridedArrayTag> volume, vigra::NumpyArray<3u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object histogramRange='globalminmax', int binCount=64, boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<2u, vigra::Singleband<float>, vigra::StridedArrayTag> image, vigra::NumpyArray<2u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object histogramRange='globalminmax', int binCount=64, boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<3u, vigra::TinyVector<float, 3>, vigra::StridedArrayTag> volume, vigra::NumpyArray<3u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<2u, vigra::TinyVector<float, 3>, vigra::StridedArrayTag> image, vigra::NumpyArray<2u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<4u, vigra::Multiband<float>, vigra::StridedArrayTag> volume, vigra::NumpyArray<3u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
    extractRegionFeatures(vigra::NumpyArray<3u, vigra::Multiband<float>, vigra::StridedArrayTag> image, vigra::NumpyArray<2u, vigra::Singleband<unsigned int>, vigra::StridedArrayTag> labels, boost::python::api::object features='all', boost::python::api::object ignoreLabel=None)
```
</details>

## Generic Improvements

1. Development conda environment configuration
2. Changing Wiki links to direct users to GitHub Pages documentation in both interfaces
3. Minor changes include renaming `load_config` to `load_core_config` in GUI logic to reduce confusion and improve consistency in function names

## Related issues 

- #183 
- #173 
- #165 
- #162 
- #140

### CODE REVIEW

**Comment by qin-yu** [comment]
@lorenzocerrone do you want to have a look at the changes, or should I just merge it?

**Comment by qin-yu** [comment]
I think I finished improving both Legacy GUI and Napari pipeline.

**Comment by wolny** [comment]
Great work @qin-yu! Two comments from my side:
1. vigra problem should be easy to fix by using the proper types of the arguments, maybe something changed in the latest version of elf...
2. Can't we make default value for `channel` to be `0` instead of `-1` in the legacy UI?

otherwise LGTM

File: plantseg/viewer/widget/utils.py [hunk 1]
```diff
@@ -70,6 +70,54 @@ def on_done(result):
     return future
 
 
+def start_prediction_threading_process(func: Callable,
```
Comment (wolny) [inline plantseg/viewer/widget/utils.py:73]: minor: rename to either `start_prediction_thread` or `start_prediction_process` but don't use both

**Review by wolny — APPROVED** [review]

**Comment by qin-yu** [comment]
> 2. Can't we make default value for `channel` to be `0` instead of `-1` in the legacy UI?

Thanks for the review @wolny, if channel is set then `data = data[channel]`, where data is just the data in it's full dimension. If the user doesn't set channel, then the whole dataset is loaded. i.e. If `channel = 0` is the default then we need another checkbox for users to say if they have a multi-channel dataset at all. Usually users' data is just in ZYX or YX.

I'll do the merge now.

**Pull request merged**